src/main/java/com/netchill/services/streaming/TrailerService.java

```diff
@@ -18,27 +19,21 @@ private TrailerService(TrailerApiService trailerApiClient) {
         this.trailerApiClient = trailerApiClient;
     }
 
-    public TrailerKey getSerieTrailerById(Long id) {
-        Trailer trailer = trailerApiClient.getSerieTrailerById(id);
+    public Optional<Trailer> getTrailerBySerieId(Long id) {
+        Trailers trailer = trailerApiClient.getTrailerBySerieId(id);
         return this.getTrailer(trailer.getKeyList());
     }
 
-    public TrailerKey getMovieTrailerById(Long id) {
-        Trailer trailer = trailerApiClient.getMovieTrailerById(id);
+    public Optional<Trailer> getTrailerByMovieId(Long id) {
+        Trailers trailer = trailerApiClient.getTrailerByMovieId(id);
         return this.getTrailer(trailer.getKeyList());
     }
 
-    private TrailerKey getTrailer(List<TrailerKey> trailerKeys) {
-        if (trailerKeys.isEmpty()) {
-            return new TrailerKey();
-        }
-
-        for (TrailerKey key : trailerKeys) {
-            if (key.getType().equals(TRAILER_TYPE)) {
-                return key;
-            }
-        }
-
-        return trailerKeys.get(0);
+    private Optional<Trailer> getTrailer(List<Trailer> trailers) {
+        return Optional.of(trailers.stream()
+                .filter(trailer -> trailer.getType().equals(TRAILER_TYPE))
```

Comment:
il vaut mieux faire l'inverse généralement : 

```suggestion
                .filter(trailer -> TRAILER_TYPE.equals(trailer.getType()))
```

Ainsi si `trailer.getType()` est null tu n'as pas de NPE

Reply:
NPE?

Reply:
null pointer exception?

Reply:
Ahh oui j'ai compris